lib/ansiblelint/rules/PackageHasRetryRule.py

```diff
@@ -67,6 +67,16 @@ class PackageHasRetryRule(AnsibleLintRule):
         "zypper",
     ]
 
+    _module_ignore_states = [
+        "absent",
+    ]
+
     def matchtask(self, file, task):
-        return (task['action']['__ansible_module__'] in self.package_modules
-                and 'until' not in task)
+        module = task["action"]["__ansible_module__"]
+        if module in self._package_modules:
```

Comment:
Another approach to improve readability would be to have "named conditions":
```suggestion
        is_module_package = module in self._package_modules
        is_task_retryable = 'until' in task
        is_state_whitelisted = task['action'].get('state') in self._module_ignore_states
        return is_module_package and not is_task_retryable and is_state_whitelisted
```
Flat and simple.